test/unit-test/ota_utest.c

```diff
@@ -3292,3 +3296,37 @@ void test_OTA_packetsProcessedOverflow()
         TEST_ASSERT_EQUAL( pFileBlock[ idx % sizeof( pFileBlock ) ], pOtaFileBuffer[ idx ] );
     }
 }
+
+/* This test is to check if the library handles the situation where the length of the
+ * job name downloaded using OTA is greater than the maximum size supported by the library. */
+void test_OTA_jobIdMaxLength()
+{
+    pOtaJobDoc = JOB_DOC_INVALID_JOB_ID_LEN_MAX;
+
+    otaGoToState( OtaAgentStateWaitingForJob );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+
+    otaReceiveJobDocument();
+    receiveAndProcessOtaEvent();
```

Comment:
For future: It would be good if the function can return an error so that just the indented functionality can be tested here.